app/src/main/java/org/lineageos/glimpse/flow/LocationsFlow.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * SPDX-FileCopyrightText: 2023 The LineageOS Project
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.lineageos.glimpse.flow
+
+import android.content.Context
+import android.location.Address
+import android.location.Geocoder
+import android.os.Build
+import androidx.exifinterface.media.ExifInterface
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.launch
+import org.lineageos.glimpse.models.Media
+import org.lineageos.glimpse.utils.MediaStoreBuckets
+import java.util.Collections
+
+class LocationsFlow(context: Context) {
+    // Flows
+    private val mediaFlow = MediaFlow(context, MediaStoreBuckets.MEDIA_STORE_BUCKET_REELS.id)
+
+    // Content resolver
+    private val contentResolver = context.contentResolver
+
+    // Geocoder
+    private val geocoder = Geocoder(context)
+
+    // Coroutines
+    @OptIn(ExperimentalCoroutinesApi::class)
+    private val ioScope = CoroutineScope(Job() + Dispatchers.IO.limitedParallelism(5))
+
+    /**
+     * Get a flow of location to media, to use as a thumbnail.
+     */
+    fun get() = callbackFlow {
```

Comment:
should use channelFlow, callbackFlow will crash is job gets cancelled and awatClose is never called ( or doesn't exist I assume )